crates/rbuilder/src/backtest/fetch/data_source.rs

```diff
@@ -2,14 +2,18 @@ use crate::backtest::{BuiltBlockData, OrdersWithTimestamp};
 use alloy_primitives::B256;
 use async_trait::async_trait;
 
+#[derive(Debug, Clone)]
+pub struct DatasourceData {
+    pub orders: Vec<OrdersWithTimestamp>,
+    pub built_block_data: Option<BuiltBlockData>,
```

Comment:
It can be just a function(or trait with one funciton) that you pass into the fetcher that will give you all these things at once. The default implementation will use mempool dumpster + optional flashbots db as part of one struct to implement this function.